### PR TITLE
disallows exponential operator with BigInt for targets lower than ES2016

### LINKED ISSUE #30398: Can't exponentiate BigInts
In normal JavaScript, this works `BigInt(2) ** BigInt(2)`

But writing the above in TypeScript, compiles to `Math.pow(BigInt(2), BigInt(2))`, which doesn't work.

According to MDN:

> [BigInt] cannot be used with methods in the built-in Math object

---

TypeScript code

```ts
BigInt(2) ** BigInt(2)
```

**Expected behavior:**
Return 4.

**Actual behavior:**

```
TypeError: Cannot convert a BigInt value to a number
    at Math.pow (<anonymous>)
```

### PR DESCRIPTION
**Pull request by Leko**
Fixes #30398

### CODE REVIEW

File: src/compiler/diagnosticMessages.json [hunk 1]
```diff
@@ -1051,6 +1051,10 @@
         "category": "Error",
         "code": 1360
     },
+    "A bigint literal cannot use exponential notation lower than ES2016.": {
```
Comment (ajafff) [inline src/compiler/diagnosticMessages.json]: this error is not limited to literals. consider removing "literal" from the message

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -26825,6 +26825,11 @@ namespace ts {
                                 case SyntaxKind.GreaterThanGreaterThanGreaterThanToken:
                                 case SyntaxKind.GreaterThanGreaterThanGreaterThanEqualsToken:
                                     reportOperatorError();
+                                    break;
+                                case SyntaxKind.AsteriskAsteriskToken:
```
Comment (ajafff) [inline src/compiler/checker.ts]: This error should be on `SyntaxKind.AsteriskAsteriskEqualsToken` as well. So the following code should not be allowed:

```ts
a **= 2;
```

File: src/compiler/diagnosticMessages.json (hunk 1 above)
Comment (DanielRosenwasser) [reply src/compiler/diagnosticMessages.json]: Can you move this lower down where other semantic errors are? Also, consider

`Exponentiation cannot be performed on 'bigint' values unless the 'target' option is set to 'es2016' or later.`

**Comment by Leko** [comment]
@ajafff @DanielRosenwasser Thank you for the review! I fixed code problems.

**Comment by elibarzilay** [comment]
(Rebased and updated)

**Pull request merged**